clusterctl/clusterdeployer/clusterdeployer_test.go

```diff
@@ -578,10 +643,54 @@ func TestDeleteBasicScenarios(t *testing.T) {
 	}
 }
 
-func generateMachines() []*clusterv1.Machine {
+func generateMasterNode(name string) *clusterv1.Machine {
 	master := &clusterv1.Machine{}
-	master.Name = "test-master"
+	master.Name = name
```

Comment:
It was this way before but I don't understand why these parameters are set here rather than when creating the Machine on the above line. Is there a reason we can't write
```
  return &clusterv1.Machine{
    Name: name,
    Spec: clusterv1.MachineSpec{
      Versions: clusterv1.MachineVersionInfo{
        ControlPlane: "1.10.1",
      },
    },
  }
```

(or the variation of that which compiles)?

Reply:
No reason. I was just following the way this was before. Now I've moved the setting of fields into the initializer.